File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.java
class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        
        for (int j = 0; j < section.length; ) {
            answer++; 
            int start = section[j]; 
            // 롤러가 덮을 수 있는 최대 구역까지 인덱스를 증가
            while (j < section.length && section[j] < start + m) {
                j++;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.js
function solution(n, m, section) {
    let cnt = 1;
    let start = section[0];
    
    for(let i=0; i<section.length; i++){
        if(start + m <= section[i]){
            cnt++;
            start = section[i];
        }
    }
    return cnt;
}